force-app/main/default/classes/tasks/CollectionsTasks.cls

```diff
@@ -18,15 +24,39 @@ public class CollectionsTasks {
     Integer sum,
     List<Integer> listWithElementsToCheck
   ) {
-    return false;
+    Boolean result = false;
+    for (Integer i = 0; i < listWithElementsToCheck.size(); i++) {
+      for (Integer j = 0; j < listWithElementsToCheck.size(); j++) {
+        if (
+          i <> j &&
+          listWithElementsToCheck[i] + listWithElementsToCheck[j] == sum
+        ) {
+          result = true;
+          break;
+        }
+      }
+    }
+    return result;
   }
 
+
   // use single loop and an arbitrary collection to address same requirements as in sumCanBeReached
   public static Boolean sumCanBeReachedOptimal(
     Integer sum,
     List<Integer> listWithElementsToCheck
   ) {
-    return false;
+    Boolean b = false;
```

Comment:
I checked this implementation on the datasets from the previous tasks and it failed. Took me a while to understand the trick though :)

This implementation looks like a hack for the test data - @baksale is working on the new set of test data to make this hack unavailable. Nice try though :)

Reply:
I was re-write this method